2996_reflection/reflection.md

```diff
@@ -3158,47 +3145,37 @@ in the *id-expression* of a class member access expression ([expr.ref]). [...]
 
-Add a bullet to paragraph 4, and renumber accordingly:
-
-::: std
-
-[4]{.pnum} An unnamed namespace or a namespace declared directly or indirectly within an unnamed namespace has internal linkage. All other namespaces have external linkage. The name of an entity that belongs to a namespace scope that has not been given internal linkage above and that is the name of
-
-* [4.1]{.pnum} a variable; or
-* [4.2]{.pnum} a function; or
-
-...
-
-has its linkage determined as follows:
-
-* [4.7]{.pnum} if the enclosing namespace has internal linkage, the name has internal linkage;
-* [4.8]{.pnum} otherwise, if the declaration of the name is attached to a named module ([module.unit]) and is not exported ([module.interface]), the name has module linkage;
-* [4.9]{.pnum} [otherwise, if the declaration introduces a variable whose type is a consteval-only type ([basic.types.general]) or a class template specialization where one or more non-type template arguments are of consteval-only type, the name has internal linkage.]{.addu}
-* [4.10]{.pnum} otherwise, the name has external linkage.
-
-:::
-
 Add a bullet to paragraph 13:
 
 ::: std
 
 [13]{.pnum} A declaration `$D$` _names_ an entity `$E$` if
 
 * [13.1]{.pnum} `$D$` contains a _lambda-expression_ whose closure type is `$E$`,
-* [13.1+]{.pnum} [`$D$` contains a _splice-specifier_ designating `$E$`,]{.addu}
+* [13.1+]{.pnum} [`$D$` contains an expression that represents either `$E$` or a `$typedef-name$` or `$namespace-alias$` that denotes `$E$`,]{.addu}
 * [13.2]{.pnum} `$E$` is not a function or function template and `$D$` contains an *id-expression*, *type-specifier*, *nested-name-specifier*, *template-name*, or *concept-name denoting* `$E$`, or
 * [13.#]{.pnum} `$E$` is a function or function template and `$D$` contains an expression that names `$E$` ([basic.def.odr]) or an *id-expression* that refers to a set of overloads that contains `$E$`.
 
 :::
 
+[The below addition of "value or object of a TU-local type" is in part a drive-by fix to make sure that enumerators in a TU-local enumeration are also TU-local]{.ednote}
+
 Extend the definition of _TU-local_ values and objects to include reflections:
 
 ::: std
 
 [16]{.pnum} A value or object is _TU-local_ if either
 
 * [16.1]{.pnum} it is, or is a pointer to, a TU-local function or the object associated with a TU-local variable, [or]{.rm}
-* [[16.1+]{.pnum} it is a reflection representing a TU-local entity, or]{.addu}
+
+:::addu
+* [16.1a]{.pnum} it is a value or object of a TU-local type,
+* [16.1b]{.pnum} it is a reflection representing
+  * [16.1b.#]{.pnum} a local entity,
+  * [16.1b.#]{.pnum} a TU-local value or object,
+  * [16.1b.#]{.pnum} a `$typedef-name$`, namespace alias, or base specifier introduced by a declaration that is either an exposure or declared at block scope, or
+  * [16.1b.#]{.pnum} an enumerator or a description of a declaration of a nonstatic data member whose type is TU-local, or
+:::
```

Comment:
This is to lift the linkage restriction right?

Could the reflection of an enumerator be considered a TU-local value (and thus we could simplify that here)?

Reply:
Are you suggesting that all reflections of enumerators are TU-local values? If so, I'm not sure why we'd want that tighter restriction.